From: Aurelien <aurelien@spartian-1>
Date: Sat, 11 Apr 2026 00:00:00 +0200
Subject: [PATCH] cmake: add missing cstdint include for GCC15 C++23

Upstream-Status: Backport [cmake 3.31+]
Signed-off-by: Aurelien <aurelien@spartian-1>
---
--- a/Utilities/cmcppdap/include/dap/network.h.orig	2026-04-12 08:48:19.130009341 +0200
+++ b/Utilities/cmcppdap/include/dap/network.h	2026-04-12 08:48:19.136821164 +0200
@@ -14,6 +14,7 @@
 
 #ifndef dap_network_h
 #define dap_network_h
+#include <cstdint>
 
 #include <functional>
 #include <memory>
