From: Aurélien DESBRIERES <aurelien@hackers.camp>
Date: Sat, 11 Apr 2026 00:00:00 +0200
Subject: [PATCH] 0001-slurm-fix-all-plugins-cross-compile-undefined-symbol

Upstream-Status: Submitted [https://support.schedmd.com/]
Signed-off-by: Aurélien DESBRIERES <aurelien@hackers.camp>
---
--- a/src/plugins/mpi/pmix/Makefile.am	2026-04-11 09:11:07.850304449 +0200
+++ src/plugins/mpi/pmix/Makefile.am	2026-04-11 09:11:07.858300800 +0200
@@ -23,7 +23,7 @@
 pmix_src += pmixp_dconn_ucx.c pmixp_dconn_ucx.h
 endif
 
-pmix_ldflags = $(PLUGIN_FLAGS) $(HWLOC_LDFLAGS) $(UCX_LDFLAGS)
+pmix_ldflags = $(PLUGIN_FLAGS) -Wl,--allow-shlib-undefined $(HWLOC_LDFLAGS) $(UCX_LDFLAGS)
 pmix_libadd = $(HWLOC_LIBS) $(UCX_LIBS)
 
 # pkglib_v#dir is needed in each one of the below because of the way we
--- a/src/plugins/acct_gather_profile/hdf5/Makefile.am	2026-04-11 09:11:07.858300800 +0200
+++ src/plugins/acct_gather_profile/hdf5/Makefile.am	2026-04-11 09:11:07.861550160 +0200
@@ -24,7 +24,7 @@
 libhdf5_api_la_LIBADD = $(HDF5_LIBS)
 
 acct_gather_profile_hdf5_la_SOURCES = $(HDF5_SOURCES)
-acct_gather_profile_hdf5_la_LDFLAGS = \
+acct_gather_profile_hdf5_la_LDFLAGS = -Wl,--allow-shlib-undefined \
 	$(PLUGIN_FLAGS) $(HDF5_LDFLAGS)
 acct_gather_profile_hdf5_la_LIBADD = $(HDF5_LIBS) libhdf5_api.la
 
