From: Aurélien DESBRIERES <aurelien@hackers.camp>
Date: Sat, 11 Apr 2026 00:00:00 +0200
Subject: [PATCH] 0002-slurm-disable-default-prep-script-plugin

Upstream-Status: Submitted [https://support.schedmd.com/]
Signed-off-by: Aurélien DESBRIERES <aurelien@hackers.camp>
---
--- a/src/common/read_config.h	2026-04-11 09:11:39.363638134 +0200
+++ src/common/read_config.h	2026-04-11 09:11:39.370304800 +0200
@@ -126,7 +126,7 @@
 #else
 #  define DEFAULT_PROCTRACK_TYPE      "proctrack/pgid"
 #endif
-#define DEFAULT_PREP_PLUGINS        "prep/script"
+#define DEFAULT_PREP_PLUGINS ""
 #define DEFAULT_PRIORITY_DECAY      604800 /* 7 days */
 #define DEFAULT_PRIORITY_CALC_PERIOD 300 /* in seconds */
 #define DEFAULT_PRIORITY_TYPE       "priority/multifactor"
