--- a/src/slurmctld/Makefile.am.orig	2026-04-10 13:36:05.143513378 +0200
+++ b/src/slurmctld/Makefile.am	2026-04-10 13:36:05.147877963 +0200
@@ -80,7 +80,7 @@
 sbin_PROGRAMS = slurmctld
 
 slurmctld_LDADD = $(LIB_SLURM) $(SLURMCTLD_INTERFACES)
-slurmctld_LDFLAGS = $(CMD_LDFLAGS)
+slurmctld_LDFLAGS = $(CMD_LDFLAGS) -Wl,--export-dynamic
 
 slurmctld_DEPENDENCIES = $(LIB_SLURM_BUILD) $(SLURMCTLD_INTERFACES)
 
