From: Aurélien DESBRIERES <aurelien@hackers.camp>
Date: Sat, 11 Apr 2026 00:00:00 +0200
Subject: [PATCH] 0003-slurm-slurmctld-export-dynamic

Upstream-Status: Submitted [https://support.schedmd.com/]
Signed-off-by: Aurélien DESBRIERES <aurelien@hackers.camp>
---
--- a/src/slurmctld/Makefile.am	2026-04-11 09:11:39.380304800 +0200
+++ src/slurmctld/Makefile.am	2026-04-11 09:11:39.386971467 +0200
@@ -75,7 +75,7 @@
 dependencies = $(SLURMCTLD_INTERFACES) $(top_builddir)/src/stepmgr/libstepmgr.la
 
 slurmctld_LDADD = $(LIB_SLURM) $(dependencies)
-slurmctld_LDFLAGS = $(CMD_LDFLAGS)
+slurmctld_LDFLAGS = $(CMD_LDFLAGS) -Wl,--export-dynamic
 
 slurmctld_DEPENDENCIES = $(LIB_SLURM_BUILD) $(dependencies)
 
