--- a/src/common/plugin.c	2026-04-10 14:06:32.773533205 +0200
+++ slurm-23.11.7/src/common/plugin.c	2026-04-10 14:06:32.779488072 +0200
@@ -123,7 +123,7 @@
 	plugin_handle_t plug;
 
 	(void) dlerror();
-	if (!(plug = dlopen(fq_path, RTLD_LAZY))) {
+	if (!(plug = dlopen(fq_path, RTLD_LAZY | RTLD_GLOBAL))) {
 		debug3("%s: dlopen(%s): %s", __func__, fq_path, dlerror());
 		return ESLURM_PLUGIN_DLOPEN_FAILED;
 	}
@@ -152,7 +152,7 @@
 	 *
 	 */
 	(void) dlerror();
-	plug = dlopen(fq_path, RTLD_LAZY);
+	plug = dlopen(fq_path, RTLD_LAZY | RTLD_GLOBAL);
 	if (plug == NULL) {
 		error("plugin_load_from_file: dlopen(%s): %s",
 		      fq_path, dlerror());
