From: Aurelien <aurelien@spartian-1>
Date: Sat, 11 Apr 2026 00:00:00 +0200
Subject: [PATCH] 0004-slurm-plugin-use-rtld-global

Upstream-Status: Submitted [https://support.schedmd.com/]
Signed-off-by: Aurelien <aurelien@spartian-1>
---
--- a/src/common/plugin.c	2026-04-11 09:11:39.393638134 +0200
+++ src/common/plugin.c	2026-04-11 09:11:39.402401053 +0200
@@ -122,7 +122,7 @@
 	plugin_handle_t plug;
 
 	(void) dlerror();
-	if (!(plug = dlopen(fq_path, RTLD_LAZY))) {
+	if (!(plug = dlopen(fq_path, RTLD_LAZY | RTLD_GLOBAL))) {
 		debug3("%s: dlopen(%s): %s", __func__, fq_path, dlerror());
 		return ESLURM_PLUGIN_DLOPEN_FAILED;
 	}
@@ -152,7 +152,7 @@
 	 *
 	 */
 	(void) dlerror();
-	plug = dlopen(fq_path, RTLD_LAZY);
+	plug = dlopen(fq_path, RTLD_LAZY | RTLD_GLOBAL);
 	if (plug == NULL) {
 		if (required)
 			error("plugin_load_from_file: dlopen(%s): %s",
